CLDR-16665 Add documentation of 'core' type values (#4727)
unicode-org/cldr: docs/site/translation/displaynames/locale-option-names-key.md

---
title: 'Locale Option Names'
---

# Locale Option Names

> Locale codes can have special variants, to indicate the use of particular calendars, or other features. They be used to select among different options in menus, and also display which options are in effect for the user. The locale codes use a special format like `de_DE`; the names for those locales are assembled from names for language, script, region, and so on. Locale codes can also carry options, such as when users want to use native digits or ASCII digits (aka Latin digits). The full name of a locale will include those options in a short format, such as _German (Germany, Buddhist Calendar)_.

Reviewer: ```suggestion
Locale codes can have special variants, to indicate the use of particular calendars or other features. They can be used to select among different options in menus, and also display which options are in effect for the user. The locale codes use a special format like `de_DE`; the names for these locales are assembled from names for language, script, region, and so on. Locale codes can also carry options, such as when users want to use native digits or ASCII digits (aka Latin digits). The full name of a locale will include those options in a short format, such as _German (Germany, Buddhist Calendar)_.
```

> When displayed as a menu or in certain other contexts, the name of the _key_ for the option (such as _Calendar_) is split from the names of the different _values_ for that option (such as _Buddhist, Gregorian, Japanese_, etc.) That shorter version is called the core value.

Reviewer: ```suggestion
When displayed as a menu or in certain other contexts, the name of the _key_ for the option (such as _Calendar_) is split from the names of the different _values_ for that option (such as _Buddhist, Gregorian, Japanese_, etc.). That shorter version is called the core value.
```

> So you will be asked to translate the long name (such as _Buddhist Calendar_), plus the name of each key (_Calendar_) and the name of the "core" value (_Buddhist_). This allows systems to show users of your language what all the locale options are.

Reviewer: ```suggestion
You will be asked to translate the long name (such as _Buddhist Calendar_), plus the name of each key (_Calendar_) and the name of the "core" value (_Buddhist_). This allows systems to show users of your language what all the locale options are.
```

| Code | Name |
| -- | -- |
| `calendar` | Kalender |
| `calendar-buddhist` | Buddhistischer Kalender |
| `calendar-buddhist-core` | Buddhistischer |

> The core name is typically used in menu listings or pull-downs, where the key name is used as the header, such as:

Reviewer: ```suggestion
The core name is typically used in menu listings or pull-down menus where the key name is used as the header, such as:
```

    - **Calendar**
        - Buddhist
        - Chinese
        - Coptic
        - …
     
This avoids the repetition that you would see if the long name were used:
    - **Calendar**
        - Buddhist Calendar
        - Chinese Calendar
        - Coptic Calendar
        - …

The core name can aso be used as an alternate form of a full locale name, such as "English (UK, Calendar: Buddhist)".

## Locale Option Names

Here are examples of names of Options to be translated.

| Key | Meaning   |
|---|---|
| Calendar | Calendar system (the European calendar is called "Gregorian"; others are the Chinese Lunar Calendar, and so on.) |
| Collation | How text is sorted (where a language has different possible ways to sort). |

> | Currency | The default currency. (The value is any currency value, such as USD). |

Reviewer: ```suggestion
| Currency | The default currency. (The value is any currency value, such as USD.) |
```

> | Numbers | The numbering system in use, such as European (0,1,2), Arabic (٠, ١, ٢ ), Devanagari ( ०,  १,  २ ). <br /> - Usually these are just derived from the name of the script.<br /> - There are some special forms, such as " Simplified Chinese Financial Numerals " or " Full Width Digits ". |

Reviewer: ```suggestion
| Numbers | The numbering system in use, such as European (0,1,2), Arabic (٠, ١, ٢ ), Devanagari ( ०,  १,  २ ). <br /> - Usually these are just derived from the name of the script.<br /> - There are some special forms, such as "Simplified Chinese Financial Numerals" or "Full Width Digits". |
```

## Full List

> The following provides a full list of keys and values. You may see just a subset of the keys and/or their values, depending on your coverage level. (It does exclude the Transform options, because those are lesser-used.)

Reviewer: ```suggestion
The following provides a full list of keys and values. You may see just a subset of the keys and/or their values, depending on your coverage level. (It does exclude the Transform options, because those are used less often.)
```

> Note that in many cases the values do not need to be translated, because the data is available elsewhere. For example, most numbering systems ar identified by their Script code. Those are indicated in the list below by all-uppercase Codes.

Reviewer: ```suggestion
Note that in many cases the values do not need to be translated, because the data is available elsewhere. For example, most numbering systems are identified by their Script code. Those are indicated in the list below by all-uppercase Codes.
```

|islamic-rgsa|Hijri, Saudi Arabia sighting|Hijri calendar, Saudi Arabia sighting|
|islamic-tbla|Hijri (tabular, astronomical epoch)|Hijri calendar, tabular (intercalary years [2,5,7,10,13,16,18,21,24,26,29] - astronomical epoch)|
|islamic-umalqura|Hijri (Umm al-Qura)|Hijri calendar, Umm al-Qura|
|iso8601|Gregorian YMD|Gregorian calendar, but all fields in descending order: era, year, month, day, day-of-week, hour, minute, second|
|japanese|Japanese|Japanese Imperial calendar|
|persian|Persian|Persian calendar|
|roc|Minguo|Republic of China calendar|

#### First day of week  (fw)
First day of week

|Codes|English Name|Description|
|-|-|-|
|DAY_OF_WEEK|_translation by other CLDR data_|Any day of the week|

#### Hour Cycle (12 vs 24)  (hc, hours)
Hour cycle (12 vs 24)

|Codes|English Name|Description|
|-|-|-|
|h11|12 (0–11)|Hour system using 0–11; 'K' in patterns|
|h12|12 (1–12)|Hour system using 1–12; 'h' in patterns|
|h23|24 (0–23)|Hour system using 0–23; 'H' in patterns|
|h24|24 (1–24)|Hour system using 1–24; 'k' in patterns|
### Sorting

#### Sort Order  (co, collation)

> Collation type

Reviewer 2: ```suggestion
Collation type

The [introduction section of the Unicode Collation Algorithm report](https://www.unicode.org/reports/tr10/#Introduction) is a good description of how collation works in software.
```
